pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,431 @@
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+# Author: Karl  Schopmeyer <inovadevelopment.com>
+#
+
+"""
+Implementation of the BaseRepository abstract classes that defines an in-memory
+repository based on dictionaries.
+"""
+
+from __future__ import absolute_import, print_function
+
+import six
+
+from pywbem import CIMClass, CIMQualifierDeclaration, CIMInstance
+
+from pywbem._nocasedict import NocaseDict
+from pywbem._utils import _format
+
+from ._baserepository import ObjectStoreAPI, BaseRepositoryAPI
+from ._utils import _uprint
+
+__all__ = ['InMemoryRepository']
+
+
+class InMemoryObjStore(ObjectStoreAPI):
+    """
+    Derived from the :class:`ObjectStoreAPI`, this class implements a
+    dictionary based in-memory repository for CIM objects that manages
+    CIM classes, CIM instances, and CIM qualifier declarations.
+    """
+
+    def __init__(self, case_insensitive_names, cim_object_type):
+        """
+          Parameters:
+
+            case_insensitive_names(:class:`py:bool`):
+                Defines whether the data store for this type handles
+                case insesitive names or not
+
+            cim_object_type ():
+             CIMType of the object stored in this object store. Only CIMClass,
+             CIMInstance, and CIMQualifierDeclaration types are allowed.
+        """
+
+        super(InMemoryObjStore, self).__init__(case_insensitive_names,
+                                               cim_object_type)
+
+        self._data = NocaseDict() if case_insensitive_names else {}
+
+    def __repr__(self):
+        return _format('InMemoryObjStore(type={0},  dict={1}, size={2}',
+                       self._cim_object_type, type(self._data),
+                       len(self._data))
+
+    def exists(self, name):
+        """
+        Overrides the abstract method.
+
+        Returns boolean True if the object is in the dictionary. Otherwise
+        returns False
+        """
+        return name in self._data
+
+    def get(self, name, copy=True):
+        """
+        Overrides the abstract method.
+
+        Gets the object with name from the dictionary or returns an exception
+        if an object with name does note exist.
+
+        If copy is True, it generates creates a copy and returns the copy
+        """
+
+        # pylint: disable=no-else-return
+        if name in self._data:
+            if copy:
+                return self._data[name].copy()
+            return self._data[name]
+        else:
+            raise KeyError('{} not in {} object store'
+                           .format(name, self._cim_object_type))
+
+    def create(self, name, cim_object):
+        """
+        Overrides the abstract method.
+
+        Creates a new cim_object in the object store dictionary or fails
+        with an exception if the name already exists in the dictionary.
+        """
```

Comment:
Add sections Parameters, Returns and Raises.

Reply:
See comment above